libvdsk/vdsk_int.h

```diff
@@ -60,6 +61,63 @@ struct vdsk_format {
 SET_DECLARE(libvdsk_formats, struct vdsk_format);
 #define	FORMAT_DEFINE(nm)	DATA_SET(libvdsk_formats, nm)
 
+/* QCOW HEADER */
```

Comment:
Can you keep this outside of vdsk_int.h?

It's not scalable to add scheme-specific definitions to a common header. Either we should use a generic indirection (like void *aux you added), or we should allow the probe function to realloc() the vdsk structure so that we can do something like:

struct qcdsk {
    struct vdsk vdsk;
    /* QCOW specific data */
};

Thoughts?

Reply:
You are right. Those structures should not be in there.

Both approaches sound ok to me. I think the first one would be easier to follow from a reader's point of view.

Which one do you suggest I use?

Reply:
I think I prefer to avoid the extra indirection. I just don't know exactly what it takes and thus how much easier the aux pointer ends up being. We'll figure it out. Let me pull this first...